tests/test_abort.js

```diff
@@ -1,157 +1,180 @@
 
-  const req = http.get('http://example.test/')
+    const req = http.request('http://example.test')
+    const emitSpy = sinon.spy(req, 'emit')
 
-  req.once('error', err => {
-    // This is the expected first error event emitted, triggered by
-    // `req.abort()`.
-    expect(err.code).to.equal('ECONNRESET')
+    req.on('error', () => {}) // listen for error so it doesn't bubble
+    req.on('socket', () => {
+      req.abort()
+      req.abort()
+      req.abort()
+    })
+    req.end()
 
-    req.once('error', err => {
-      // This is the abort error under test, triggered by `req.end()`
-      expect(err.message).to.equal('Request aborted')
-      scope.done()
+    setTimeout(() => {
+      const events = emitSpy.args.map(i => i[0])
+      expect(events).to.deep.equal(['socket', 'abort', 'error', 'close'])
+      expect(scope.isDone()).to.be.false()
       done()
-    })
+    }, 10)
   })
 
-  process.nextTick(() => req.abort())
-  process.nextTick(() => req.end())
-})
+  it('should emit an ECONNRESET error when aborted inside a `finish` event listener', done => {
+    const scope = nock('http://example.test').get('/').reply()
 
-it('`req.flushHeaders()` on an aborted request should trigger the expected error', done => {
-  const scope = nock('http://example.test').get('/').reply()
+    const req = http.request('http://example.test')
+    const emitSpy = sinon.spy(req, 'emit')
 
-  const req = http.get('http://example.test/')
+    req.on('finish', () => {
+      req.abort()
+    })
+    req.on('error', err => {
+      expect(err.message).to.equal('socket hang up')
+      expect(err.code).to.equal('ECONNRESET')
+    })
+    req.end()
+
+    setTimeout(() => {
+      const events = emitSpy.args.map(i => i[0])
+      expect(events).to.deep.equal([
+        'socket',
+        'finish',
+        'abort',
+        'error',
+        'close',
+      ])
+      expect(scope.isDone()).to.be.false()
+      done()
+    }, 10)
+  })
 
-  req.once('error', err => {
-    // This is the expected first error event emitted, triggered by
-    // `req.abort()`.
-    expect(err.code).to.equal('ECONNRESET')
+  // The Interceptor is considered consumed just prior to the `response` event on the request,
+  // all tests below assert the Scope is done.
 
-    req.once('error', err => {
-      // This is the abort error under test, triggered by `req.flushHeaders()`
-      expect(err.message).to.equal('Request aborted')
+  it('should not emit an error when called inside a `response` event listener', done => {
+    const scope = nock('http://example.test').get('/').reply()
+
+    const req = http.request('http://example.test')
+    const emitSpy = sinon.spy(req, 'emit')
+
+    req.on('response', () => {
+      req.abort()
+    })
+    req.end()
+
+    setTimeout(() => {
+      const events = emitSpy.args.map(i => i[0])
+      expect(events).to.deep.equal([
+        'socket',
+        'finish',
+        'response',
+        'abort',
+        'close',
```

Comment:
As above, you're asserting the order of these events. In addition to asserting that `'error'` isn't included, s this a confidence check?